car/car.ino

```diff
@@ -143,8 +144,8 @@ void handleInput() { //handle serial input (String!!)
 void obstacleAvoidance(){
   int leftDistance = left.readRangeContinuousMillimeters();
   int rightDistance = right.getMedianDistance();
-
-  if (leftDistance > LEFT_DIST && rightDistance > RIGHT_DIST){   //if there is no obstical on either sides rotate 90 degrees to the left
+  
+  if (leftDistance > LEFT_DIST && rightDistance > RIGHT_DIST || leftDistance > LEFT_DIST && rightDistance == 0){   //if there is no obstical on either sides rotate 90 degrees to the left
```

Comment:
@platisd When the car enter the if statment (or the else if/else) the first time after turning on the car. it continues to drive into the obstacle that has mede it stop in the first place. meaning that the car stops when i has detected an obstacle but then continues to drive into the obstical. once the car has stoped and you test it again (without turning it off) stop in front of the obstical (without hitting it) and then drives around it as it should. (Like in the demo we showed you) We have printed the values from the sensors while testing and nothing unexpected have appeard.

do you have any ideas of what could be wrong?

Reply:
Responding to both @majawennroth @oscarhjern 

I can't 100% follow, but have you verified that `rotateOnSpot` works as intended when this behavior is exhibited?

The logic of the `if` statements doesn't seem wrong which agrees with the results of your investigation. On a different and unrelated note, I would suggest adding more parentheses in order to clarify which comparisons go together. ([ES.41](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-parens))
I had to look up [operator precedence](https://en.cppreference.com/w/cpp/language/operator_precedence) in C++ to make sure the expression would be evaluated the way you intend.

Anyway, it sounds like for currently unknown reasons `rotateOnSpot` it doesn't *actually* rotate the car by `TURN_LEFT` degrees. Printing out the degrees returned by the gyroscope via `car.getHeading()` will shed some more light on the situation.

Reply:
Long shot but could be a quick-fix: Add a `car.update()` in the `loop()`. As [explained in the documentation](https://platisd.github.io/smartcar_shield/class_smart_car.html#ae6cd4a31632721f3a808c36da15cd5d3) it is suggested that you have this being executed as often as possible. In your case you have it only when you want to rotate which makes me suspect that there may be a lot of error the first time you call `car.getHeading()`.

Reply:
**Update:** Car.update() worked!! @platisd